### PR TITLE
Issue #7655: Update doc for ParameterAssignment

### PR DESCRIPTION
**Pull request by AkMo3**
Closes: #7655 
Since PR #9013 was abandoned and minimal changes were required in that PR to complete, this PR completes those changes. 

Please comment if it is allowed to make this PR while on is opened and abandoned?

#### Test File
```
$ cat Test.java
class MyClass {
    int method(int parameter) {
        if (parameter <= 0 ) {
            throw new IllegalArgumentException("A positive value is expected");
        }
        parameter -= 2;  // violation
        return parameter;
    }
}

class MyClass {
    int method(int parameter) {
        if (parameter <= 0 ) {
            throw new IllegalArgumentException("A positive value is expected");
        }
        parameter -= 2;  // violation
        return parameter;
    }
}
```
#### Config.xml
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
  <module name="ParameterAssignment"/> 	
  </module>
</module>
```
#### Results with checkstyle
```
$ java -jar checkstyle-8.41-SNAPSHOT-all.jar -c config.xml Test.java                
Starting audit...
[ERROR] /home/akmo/GitHub/AbandonedPR/Test.java:6:19: Assignment of parameter 'parameter' is not allowed. [ParameterAssignment]
[ERROR] /home/akmo/GitHub/AbandonedPR/Test.java:16:19: Assignment of parameter 'parameter' is not allowed. [ParameterAssignment]
Audit done.
Checkstyle ends with 2 errors
```

### CODE REVIEW

**Comment by AkMo3** [comment]
Github, generate site

**Comment by AkMo3** [comment]
Open JDK failed due to `connection reset`, please restart it.

Log - ` Transfer failed for https://repo.maven.apache.org/maven2/com/github/sevntu-checkstyle/sevntu-checks/1.38.0/sevntu-checks-1.38.0.pom: Connection reset -> [Help 1]`

**Comment by AkMo3** [comment]
Github, generate site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java [hunk 1]
```diff
@@ -50,6 +50,30 @@
...
+ *     int method(int parameter) {
+ *         if (parameter &lt;= 0 ) {
+ *             throw new IllegalArgumentException("A positive value is expected");
+ *         }
+ *         parameter -= 2;  // violation
+ *         return parameter;
+ *     }
+ * }
+ *
+ * class MyClass {
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java:66]: why this example is duplicated?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java [hunk 2]
```diff
@@ -50,6 +50,30 @@
...
+ * </p>
+ * <pre>
+ * class MyClass {
+ *     int method(int parameter) {
+ *         if (parameter &lt;= 0 ) {
+ *             throw new IllegalArgumentException("A positive value is expected");
+ *         }
+ *         parameter -= 2;  // violation
+ *         return parameter;
+ *     }
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java:63]: please make one more example with stupid case when it is just an assignment to parameter (not -=)

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java [hunk 3]
```diff
@@ -50,6 +50,30 @@
  * &lt;module name=&quot;ParameterAssignment&quot;/&gt;
  * </pre>
  * <p>
+ * Example :
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java:53]: whitespace before colon

File: src/xdocs/config_coding.xml [hunk 4]
```diff
@@ -5836,6 +5836,30 @@ public class AnnotationLocationCheck extends AbstractCheck {
         <source>
 &lt;module name=&quot;ParameterAssignment&quot;/&gt;
         </source>
+        <p>
+            Example :
```
Comment (strkkk) [inline src/xdocs/config_coding.xml:5840]: indentation should be 2

**Review by strkkk — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java (hunk 1 above)
Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java:66]: Corrected, done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java (hunk 2 above)
Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java:60]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java (hunk 3 above)
Comment (AkMo3) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java]: Done.

File: src/xdocs/config_coding.xml (hunk 4 above)
Comment (AkMo3) [reply src/xdocs/config_coding.xml]: Done

**Comment by strkkk** [comment]
Github, generate site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java [hunk 5]
```diff
@@ -50,6 +50,31 @@
...
+ *   int method(int parameter) {
+ *     if (parameter &lt;= 0 ) {
+ *       throw new IllegalArgumentException("A positive value is expected");
+ *     }
+ *     parameter -= 2;  // violation
+ *     return parameter;
+ *   }
+ * }
+ *
+ * class MyClass {
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java:66]: there is no need for class, just move method (with different name) to first class

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java [hunk 6]
```diff
@@ -50,6 +50,31 @@
...
+ *   int method(int parameter) {
+ *     int var = parameter;
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java:68]: Please
1. rename variable (var is not a good name)
2. move it after if block (check original parameter in if condition)

**Review by strkkk — CHANGES_REQUESTED** [review]

**Review by strkkk — COMMENTED** [review]
I am ok with examples, but please squash commits into one

**Comment by AkMo3** [comment]
> I am ok with examples, but please squash commits into one

Done.

**Review by strkkk — APPROVED** [review]

**Review by pbludov — APPROVED** [review]

**Review by pbludov — APPROVED** [review]

**Review by rnveach — APPROVED** [review]

**Pull request merged**